images/controller/internal/controllers/rv_status_config_shared_secret/reconciler.go

```diff
@@ -0,0 +1,275 @@
+	}
+
+	// Check if sharedSecret is not set - generate new one
+	// Filtering through early return instead of WithEventFilter
+	if rv.Status == nil || rv.Status.DRBD == nil || rv.Status.DRBD.Config == nil || rv.Status.DRBD.Config.SharedSecret == "" {
+		return r.reconcileGenerateSharedSecret(ctx, rv, log)
+	}
+
+	// Check for UnsupportedAlgorithm errors in RVRs
+	return r.reconcileHandleUnsupportedAlgorithm(ctx, rv, log)
+}
+
+// reconcileGenerateSharedSecret generates a new shared secret and selects the first algorithm
+func (r *Reconciler) reconcileGenerateSharedSecret(
+	ctx context.Context,
+	rv *v1alpha3.ReplicatedVolume,
+	log logr.Logger,
+) (reconcile.Result, error) {
+	// Update RV status with shared secret
+	// If there's a conflict (409), return error - next reconciliation will solve it
+	// Race condition handling: If two reconciles run simultaneously, one will get 409 Conflict on Patch.
+	// The next reconciliation will check if sharedSecret is already set and skip generation.
+	from := client.MergeFrom(rv)
+	changedRV := rv.DeepCopy()
+
+	// Check if sharedSecret is already set (idempotent check after DeepCopy)
+	// Note: This check is on the copy, but if there's a race condition, Patch will return 409 Conflict
+	// and the next reconciliation will handle it correctly.
+	if changedRV.Status != nil && changedRV.Status.DRBD != nil && changedRV.Status.DRBD.Config != nil && changedRV.Status.DRBD.Config.SharedSecret != "" {
+		log.V(1).Info("sharedSecret already set and valid", "algorithm", changedRV.Status.DRBD.Config.SharedSecretAlg)
+		return reconcile.Result{}, nil // Already set, nothing to do (idempotent)
+	}
+
+	// Generate new shared secret using UUID v4 (36 characters, fits DRBD limit of 64)
+	// UUID provides uniqueness and randomness required for peer authentication
+	sharedSecret := uuid.New().String()
+	algorithm := v1alpha3.SharedSecretAlgorithms()[0] // Start with first algorithm (sha256)
+
+	log.Info("Generating new shared secret", "algorithm", algorithm)
+
+	// Initialize status if needed
+	ensureRVStatusInitialized(changedRV)
+
+	// Set shared secret and algorithm
+	changedRV.Status.DRBD.Config.SharedSecret = sharedSecret
+	changedRV.Status.DRBD.Config.SharedSecretAlg = algorithm
+
+	if err := r.cl.Status().Patch(ctx, changedRV, from); err != nil {
+		log.Error(err, "Patching ReplicatedVolume status with shared secret")
+		return reconcile.Result{}, client.IgnoreNotFound(err)
+	}
+
+	log.Info("Generated shared secret")
+	return reconcile.Result{}, nil
+}
+
+// buildAlgorithmLogFields builds structured logging fields for algorithm-related logs
+// logFields: structured logging fields for debugging algorithm operations
+func buildAlgorithmLogFields(
+	rv *v1alpha3.ReplicatedVolume,
+	currentAlg string,
+	nextAlgorithm string,
+	maxFailedIndex int,
+	maxFailedRVR *v1alpha3.ReplicatedVolumeReplica,
+	algorithms []string,
+	failedNodeNames []string,
+) []interface{} {
+	logFields := []interface{}{
```

Comment:
Это то-же самое но короче

```suggestion
) any {
	logFields := any{
```